rpcserver.go

```diff
@@ -4263,6 +4274,13 @@ func (r *rpcServer) sendPayment(stream *paymentStream) error {
 			case <-reqQuit:
 				return
 
+			case <-ctx.Done():
```

Comment:
This kinda _looks_ correct, but it is also a bit scary change to make.

I would suggest breaking this PR up a bit:
1. The lncli changes are large enough in themselves, and we can add the context awareness for a few "safe" RPCs (for instance the Subscribe calls), that don't write state.
2. A follow-up PR that adds context awareness to more RPCs can be made, when we are confident that the first context cancellation works as intended.